src/Executor/ReferenceExecutor.php

```diff
@@ -721,18 +733,22 @@ protected function resolveFieldValueOrError(
         FieldDefinition $fieldDef,
         FieldNode $fieldNode,
         callable $resolveFn,
+        callable $argsMapper,
         $rootValue,
         ResolveInfo $info,
         $contextValue
     ) {
         try {
             // Build a map of arguments from the field.arguments AST, using the
             // variables scope to fulfill any variable references.
-            $args = Values::getArgumentValues(
+            /** @phpstan-ignore-next-line */
+            $this->fieldArgsCache[$fieldDef] ??= new \SplObjectStorage();
```

Comment:
Why ignore this?

Reply:
phpstan complains that `SplObjectStorage<object, mixed>` is not assignable. I guess it defaults to template in SplObjectStorage. Seems impossible to tell that newly created `SplObjectStorage` will be `SplObjectStorage<FieldNode, mixed>`

```
            /** @var \SplObjectStorage<FieldNode, mixed> */
            $this->fieldArgsCache[$fieldDef] ??= new \SplObjectStorage();
```

 `745    PHPDoc tag @var does not specify variable name. `

Reply:
That way works, but its a bit long for typing.

```
            if (!$this->fieldArgsCache->offsetExists($fieldDef)) {
                /** @var \SplObjectStorage<FieldNode, mixed> $cache2 */
                $cache2 = new \SplObjectStorage();
                $this->fieldArgsCache->offsetSet($fieldDef, $cache2);
            }
```

I don't see big difference of either way. Its local assignment anyway - no chance of something to broke.

Reply:
Ok, you can leave the code as is - but please add a short comment to the `@phpstan-ignore` to explain that the error comes from a lack of generic inference.

Reply:
Got you.